src/Resources/Resources.Test/ScenarioTests/DeploymentScriptsTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.WindowsAzure.Commands.ScenarioTest;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Azure.Commands.Resources.Test.ScenarioTests
+{
+    public class DeploymentScriptsTests : ResourceTestRunner
+    {
+        public DeploymentScriptsTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact()]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestDeploymentScriptGetForPowerShell()
+        {
+            TestRunner.RunTestScript("Test-GetDeploymentScript-PowerShell");
+        }
+
+        [Fact()]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestDeploymentScriptGetForCli()
+        {
+            TestRunner.RunTestScript("Test-GetDeploymentScript-Cli");
+        }
+
+        [Fact(Skip = "Need to wait until the Resources .NET SDK is updated to 3.3.0 in the PS module.")]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestDeploymentScriptGetDeploymentScriptWithBadScript()
+        {
+            TestRunner.RunTestScript("Test-GetDeploymentScriptWithBadScript");
+        }
+
+        [Fact()]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestDeploymentScriptLogGetForPowerShell()
+        {
+            TestRunner.RunTestScript("Test-GetDeploymentScriptLog-PowerShell");
+        }
+
+        [Fact()]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestDeploymentScriptLogGetForCli()
+        {
+            TestRunner.RunTestScript("Test-GetDeploymentScriptLog-Cli");
+        }
+
+        [Fact(Skip = "Saving a file operation fails. To-Do Ask test infra team. ")]
```

Comment:
What's about the failure?

Reply:
The execution will not save the log file locally during. I might be missing something here. I'll reach out to you offline for this.

---
In reply to: [390161450](https://github.com/Azure/azure-powershell/pull/11266#discussion_r390161450) [](ancestors = 390161450)

Reply:
I have joined in the PS office hours today about this. I was told to implement in memory datastore mock and test against that. I'll work on that and submit this test.